src/database/import.ts

```diff
@@ -0,0 +1,80 @@
+import { URL } from "url";
+import { Client } from "../apiv2";
+
+const MAX_CHUNK_SIZE = 1024 * 1024;
+
+type Data = {
+  json: { [key: string]: any } | string | number | boolean;
+  pathname: string;
+};
+
+type ChunkedData = {
+  chunks: Data[] | null;
+  size: number;
+};
+
+/**
+ * Imports JSON data to a given RTDB instance.
+ *
+ * The data is parsed and chunked into subtrees of ~1 MB, to be subsequently written in parallel.
+ */
+export default class DatabaseImporter {
+  chunks: Data[];
+  private client: Client;
+
+  constructor(private dbUrl: URL, file: string, private chunkSize = MAX_CHUNK_SIZE) {
+    const data = { json: JSON.parse(file), pathname: dbUrl.pathname };
+    const chunkedData = this.chunkData(data);
+    this.chunks = chunkedData.chunks || [data];
+    this.client = new Client({ urlPrefix: dbUrl.origin, auth: true });
+  }
+
+  /**
+   * Writes the chunked data to RTDB. Any existing data at the specified location will be overwritten.
+   */
+  async execute(): Promise<any> {
+    return Promise.all(
```

Comment:
Yeah, we definitely want to limit the concurrency. Maybe a configurable flag that defaults to 5?

Not sure, can discuss more in the API doc.

Reply:
Done. But not configurable for now. Not sure that it's necessary.